pandas-dev/pandas
# DateOffset add to ndarray

This is WIP to implement changes needed to allow DateOffset subclasses e.g. MonthEnd to be directly added to numpy.ndarray, e.g. `numpy.array(['2022-01-01'], dtype='datetime64') + pandas.offsets.MonthEnd(1)`

I need some guidance on:

- cython mechanism to check that passed ndarray is datetime64 dtype i.e.cython equivalent of pandas.api.types.is_datetime64_any_dtype
- how to approach adding unit tests? two ideas I have
  - somehow duplicate existing offset tests but run them on ndarray instead of Series?  looks on the surface to be more difficult to implement
  - apply a set of offsets to a predefined Series and ndarray equivalent, and then check results match

Many thanks!

- [x] closes #47014
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
